bash-pgo/PKGBUILD

```diff
@@ -0,0 +1,154 @@
+# Maintainer: Levente Polyak <anthraxx[at]archlinux[dot]org>
+# Maintainer: Giancarlo Razzolini <grazzolini@archlinux.org>
+# Contributor:  Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
+# Contributor: Allan McRae <allan@archlinux.org>
+# Contributor: Aaron Griffin <aaron@archlinux.org>
+# PGO Version: Laio Seman <laio [at] ieee.org>
+
+pkgname=bash
+_basever=5.2
+_patchlevel=021
+pkgver=${_basever}.${_patchlevel}
+pkgrel=1
+pkgdesc='The GNU Bourne Again shell'
+arch=(x86_64)
+license=(GPL)
+url='https://www.gnu.org/software/bash/bash.html'
+backup=(etc/bash.bash{rc,_logout} etc/skel/.bash{rc,_profile,_logout})
+depends=(readline libreadline.so glibc ncurses)
+optdepends=('bash-completion: for tab completion')
+provides=('sh')
+install=bash.install
+source=(https://ftp.gnu.org/gnu/bash/bash-$_basever.tar.gz{,.sig}
+    dot.bashrc
+    dot.bash_profile
+    dot.bash_logout
+    system.bashrc
+    system.bash_logout)
+validpgpkeys=('7C0135FB088AAF6C66C650B9BB5869F064EA74AB') # Chet Ramey
+
+if [[ $((10#${_patchlevel})) -gt 0 ]]; then
+    for ((_p = 1; _p <= $((10#${_patchlevel})); _p++)); do
+        source=(${source[@]} https://ftp.gnu.org/gnu/bash/bash-$_basever-patches/bash${_basever//./}-$(printf "%03d" $_p){,.sig})
+    done
+fi
+
+prepare() {
+    cd $pkgname-$_basever
+
+    msg "Applying patches..."
+    for ((_p = 1; _p <= $((10#${_patchlevel})); _p++)); do
+        local patch="bash${_basever//./}-$(printf "%03d" $_p)"
+        msg2 "applying patch ${patch}"
+        patch -Np0 -i ../"${patch}"
+    done
+}
+
+build() {
+    cd $pkgname-$_basever
+
+    _bashconfig=(-DDEFAULT_PATH_VALUE=\'\"/usr/local/sbin:/usr/local/bin:/usr/bin\"\'
+        -DSTANDARD_UTILS_PATH=\'\"/usr/bin\"\'
+        -DSYS_BASHRC=\'\"/etc/bash.bashrc\"\'
+        -DSYS_BASH_LOGOUT=\'\"/etc/bash.bash_logout\"\'
+        -DNON_INTERACTIVE_LOGIN_SHELLS)
+    export CFLAGS="${CFLAGS} ${_bashconfig[@]}"
+    export CFLAGS="${CFLAGS} -fprofile-generate -fprofile-update=atomic -fprofile-partial-training"
```

Comment:
Doesnt need a path defined for GCC PGO ?
like -fprofile-generate=$PWD/PGO

Reply:
No, for GCC the default profile folder is already the binary/library path, it generates the .cgda "next to" the binary.